contracts/ClaimableNft.sol

```diff
@@ -1261,37 +57,72 @@ contract ClaimableNft is ERC721URIStorage, Ownable {
              usersMap[_beneficiary[i]]=true;
         }
      }
+    
+    // check user allowed to mint or not 
+   modifier checkIsAllowedToMint() {
+       require(usersMap[msg.sender] == true,"Not allowed to mint");
+       _;
+   }
 
-//    mint the NFT to allowed user only 
-  function mintNFT( string memory tokenURI) checkIsAllowedToMint
-       public 
-       returns (uint256)
+      function mintNFT(string memory uri)  checkIsAllowedToMint public  returns (uint256)
    {
        _tokenIds.increment();
 
        uint256 newItemId = _tokenIds.current();
-       _mint(msg.sender, newItemId);
-       _setTokenURI(newItemId, tokenURI);
-
-       mintMap[msg.sender].push(newItemId);
+       _safeMint(msg.sender, newItemId);
+       _setTokenURI(newItemId, uri);
        usersMap[msg.sender]=false;
-       
        return newItemId;
    }
+    
+    function getTokenUri(address _owner) external view returns (uint[] memory) {
 
-   function burnNFT(uint256 tokenId) public {
+        uint tokenCount = balanceOf(_owner);
+        uint[] memory tokensId = new uint256[](tokenCount);
+
+        for (uint i = 0; i < tokenCount; i++) {
+            tokensId[i] = tokenOfOwnerByIndex(_owner, i);
+        }
+        return tokensId;
+    }
+
+    function burnNFT(uint256 tokenId) public {
```

Comment:
wondering if this one should not be restricted to owner

Reply:
burn cant be done by owner of contract because user if holding the NFT so burn can only be done by users who is holding nft .

Reply:
I meant, owner of the token

if I own token #1, I should be able to burn it

like that it looks like everyone is able to do so

Reply:
yes it is like that only only owner of NFT can burn it .it checks internally that the given token id is the owner of nft or not .